cpp/src/Slice/Parser.cpp

```diff
@@ -1245,37 +1234,32 @@ Slice::Container::createModule(const string& name)
                 string msg = "module `" + name + "' is capitalized inconsistently with its previous name: `";
                 msg += module->name() + "'";
                 _unit->error(msg);
-                return 0;
+                return nullptr;
             }
         }
         else if(!differsOnlyInCase)
         {
             string msg = "redefinition of " + matches.front()->kindOf() + " `" + matches.front()->name();
             msg += "' as module";
             _unit->error(msg);
-            return 0;
+            return nullptr;
         }
         else
         {
             string msg = "module `" + name + "' differs only in capitalization from ";
             msg += matches.front()->kindOf() + " name `" + matches.front()->name() + "'";
             _unit->error(msg);
-            return 0;
+            return nullptr;
         }
     }
 
-    if(!checkIdentifier(name))
-    {
-        return 0;
-    }
-
-    ModulePtr q = new Module(this, name);
-    _contents.push_back(q);
-    return q;
+    checkIdentifier(name);
+    _unit->error("`" + name + "': module can only defined at global or module scope");
+    return nullptr;
 }
 
 ClassDefPtr
-Slice::Container::createClassDef(const string& name, int id, const ClassDefPtr& base)
+Slice::Container::createClassDef(const string& name, int, const ClassDefPtr&)
```

Comment:
I don't like all this code duplication, where `createXxxDef` on `Container` performs various checks then fails with "can only be defined at module scope", and then the `createXxxDef` override on `Module` performs the same checks and succeeds. I am sure there is a better way to do it.

Reply:
In my 2nd PR, I made a helper method for handled these, but Jose also suggested just dropping the extra checks in the Container::createX methods, what do you think?